### ISSUE TITLE
Buttons do not work properly if drawn over another button

### ISSUE THREAD

**Issue by blakat360** [ocornut/imgui#3149]
***Version/Branch of Dear ImGui:***

Version: 9f9ff84
Branch: Master

***Back-end/Renderer/Compiler/OS***

Back-ends: imgui_impl_opengl3.cpp + imgui_impl_glfw.cpp

***Issuee***

If I draw a button over another, the second button does not work and if I click it, the first button I draw captures the input.

***Code***
Consider:
```
//start frame
ImGui::SetCursorPos(0,0);
ImGui::Button("1", ImVec2(100, 100));
ImGui::SetCursorPos(0,0);
ImGui::Button("2", ImVec2(50, 50));
//end frame
```

I can't click on "2" or hover over it. When I do, it instead appears that I am hovering over and then clicking on "1".

I've looked for buttonflags that change this behaviour to no avail.

Any advice would be much appreciated.

**Comment by ocornut** [comment]
Right now this is not widely supported.

You may use:
```
ImGui::SetCursorPos(ImVec2(0, 0));
ImGui::Button("1", ImVec2(100, 100));
ImGui::SetItemAllowOverlap();
ImGui::SetCursorPos(ImVec2(0, 0));
ImGui::Button("2", ImVec2(50, 50));
```
Then you can click button 2, BUT, button 1 will show as hovered.

You can use
```
ImGui::SetCursorPos(ImVec2(0, 0));
ImGui::ButtonEx("1", ImVec2(100, 100), ImGuiButtonFlags_AllowItemOverlap);
ImGui::SetItemAllowOverlap();
ImGui::SetCursorPos(ImVec2(0, 0));
ImGui::Button("2", ImVec2(50, 50));
```
Button 1 won't show as however.

This is not public API (need imgui_internal.h) as this feature is currently quite inconsistency supported, I feel like the flag alone should suffice here without the extra function call.

**Comment by blakat360** [comment]
Issue Solved!

The flag did not suffice, and I needed to call ```SetItemAllowOverlap()```.

Good news is, button 1 still rendered which is exactly what I wanted!

Thanks :)

**Comment by ocornut** [comment]
Hello,

We made a large series of changes in 1.89.7 exactly related to this
See this release note:
https://github.com/ocornut/imgui/releases/tag/v1.89.7

TL;DR;
- This way of using `SetItemAllowOverlap()` (without imgui_internal.h) use has been broken since 1.89.0
- In 1.89.7 we introduced `SetNextItemAllowOverlap()` which is to be called BEFORE the item and it should work.
